Plugin/Tax/Config/ShippingTaxPlugin.php

```diff
@@ -0,0 +1,216 @@
+
+        if ($dynamicType === FireGentoSource::DYNAMIC_TYPE_SHIPPING_TAX_DEFAULT) {
+            return $result;  // If the default behaviour was configured -> use default tax class id
+        }
+
+        $taxClassId = false;
+        if ($dynamicType === FireGentoSource::DYNAMIC_TYPE_HIGHEST_PRODUCT_TAX) {
+            $taxClassId = $this->getHighestProductTaxClassId($quoteItems, $store);
+        }
+
+        // If no tax class id was found, use default one
+        if (!$taxClassId) {
+            return $result;
+        }
+
+        return $result->setTaxClassKey(
+            $this->taxClassKeyDataObjectFactory->create()
+                ->setType(TaxClassKeyInterface::TYPE_ID)
+                ->setValue($taxClassId)
+        );
+    }
+
+    /**
+     * Method for getting highest product tax class id
+     *
+     * @param                            $quoteItems
+     * @param null|string|bool|int|Store $store
+     *
+     * @return bool|int
+     *
+     * @throws \Magento\Framework\Exception\LocalizedException
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    private function getHighestProductTaxClassId($quoteItems, $store)
+    {
+        $taxClassIds       = [];
+        $highestTaxClassId = false;
+
+        foreach ($quoteItems as $quoteItem) {
+
+            /** @var $quoteItem Item */
+            if ($quoteItem->getParentItem()) {
+                continue;
+            }
+
+            // Retrieve the tax percent
+            $taxPercent = $quoteItem->getTaxPercent();
+            if (!$taxPercent) {
+                $taxPercent = $this->getTaxPercent($quoteItem->getTaxClassId(), $store);
+            }
+
+            // Add the tax class
+            if (($taxPercent) && !in_array($taxPercent, $taxClassIds)) {
+                $taxClassIds[$taxPercent] = (int)$quoteItem->getTaxClassId();
+            }
+        }
+
+        /**
+         * Fetch the highest tax rate
+         */
+        ksort($taxClassIds);
+        if (count($taxClassIds) > 0) {
+            $highestTaxClassId = array_pop($taxClassIds);
+        }
+        if (!$highestTaxClassId || $highestTaxClassId === null) {
+            return false;
+        }
+
+        return $highestTaxClassId;
+    }
+
+    /**
+     * Method for getting tax prcentage
+     *
+     * @param int                        $productTaxClassId
+     * @param null|string|bool|int|Store $store
+     *
+     * @return int
+     * @throws \Magento\Framework\Exception\LocalizedException
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    private function getTaxPercent(int $productTaxClassId, $store): int
+    {
+        $groupId            = $this->customerSession->getCustomerGroupId();
+        $group              = $this->groupRepository->getById($groupId);
+        $customerTaxClassId = $group->getTaxClassId();
+
+        $request = $this->taxCalculation->getRateRequest(null, null, $customerTaxClassId, $store);
```

Comment:
Are you sure this works? Isn't `$this->taxCalculation` still `null`?

Reply:
I think it was indeed broken, now this place is gone.